### PR TITLE
Explicitly set Rust Edition: 2018

### PR DESCRIPTION
**Pull request by dconnolly**

### CODE REVIEW

**Comment by hdevalence** [comment]
Looks like the build stalled for random reasons?

**Comment by dconnolly** [comment]
> Looks like the build stalled for random reasons?

I've been kicking Travis on and off and I haven't narrowed this down yet. :(

**Comment by dconnolly** [comment]
With Rust:stable moving to 1.37.0, it is now explicitly warning deprecation notices about the 2018 edition incompatibilities.

File: db/src/kv/diskdb.rs [hunk 1]
```diff
@@ -250,11 +250,11 @@ impl Database {
...
                 info!("{}", s);
                 info!("Attempting DB repair for {}", path);
-                try!(DB::repair(&opts, &path));
+                r#try!(DB::repair(&opts, &path));
 
                 match cfnames.is_empty() {
-                    true => try!(DB::open(&opts, &path)),
-                    false => try!(DB::open_cf(&opts, &path, &cfnames, &cf_options)),
+                    true => r#try!(DB::open(&opts, &path)),
+                    false => r#try!(DB::open_cf(&opts, &path, &cfnames, &cf_options)),
```
Comment (NikVolf) [inline db/src/kv/diskdb.rs:257]: Better replace those with "?" - manually or with script?

See https://users.rust-lang.org/t/why-does-cargo-fix-replace-try-with-r-try-instead-of/21972/4

**Comment by NikVolf** [comment]
What's up with this pr?
Looks like a reasonable change to merge

**Pull request closed without merging**